pkg/memory/memory_linux.go

```diff
@@ -52,10 +53,56 @@ func (i *Info) load() error {
 		i.ctx.Warn(_WARN_CANNOT_DETERMINE_PHYSICAL_MEMORY)
 		i.TotalPhysicalBytes = tub
 	}
-	i.SupportedPageSizes = memSupportedPageSizes(paths)
+	i.SupportedPageSizes, _ = memorySupportedPageSizes(paths.SysKernelMMHugepages)
 	return nil
 }
 
+func AreaForNode(ctx *context.Context, nodeID int) (*Area, error) {
+	paths := linuxpath.New(ctx)
+	path := filepath.Join(
+		paths.SysDevicesSystemNode,
+		fmt.Sprintf("node%d", nodeID),
+	)
+
+	blockSizeBytes, err := memoryBlockSizeBytes(paths.SysDevicesSystemMemory)
+	if err != nil {
+		return nil, err
+	}
+
+	totPhys, err := memoryTotalPhysicalBytesFromPath(path, blockSizeBytes)
+	if err != nil {
+		return nil, err
+	}
+
+	totUsable, err := memoryTotalUsableBytesFromPath(filepath.Join(path, "meminfo"))
+	if err != nil {
+		return nil, err
+	}
+
+	supportedHP, err := memorySupportedPageSizes(filepath.Join(path, "hugepages"))
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "err=%v\n", err)
```

Comment:
Is there a reason we are only printing to stderr on this particular error and not the other errors above? Can we call `ctx.Warn` instead for all four subcalls' error returns?

Reply:
this is because it's a debug leftover :) you raised a good point though, I'll check if/how I can add a `ctx.Warn` here.

Reply:
OK, in this case is not worth to add a `ctx.Warn` IMHO, so I'm just dropping the `Fprintf` call. However, I found some cleanups to be done and I'll add to the PR.